sklearn/utils/multiclass.py

```diff
@@ -289,8 +289,14 @@ def type_of_target(y):
             # see NEP 34
             y = np.asarray(y, dtype=object)
 
-    # The old sequence of sequences format
     try:
+        # labels in bytes format
+        if isinstance(y[0], bytes):
+            raise ValueError(
+                "Labels are represented as bytes and are not supported. "
+                "Convert the labels to Python string or integral format."
+            )
```

Comment:
Given that this is a breaking change, I agree with the comment in https://github.com/scikit-learn/scikit-learn/pull/18555#issuecomment-909006296 and deprecate the behavior. This means that we raise a warning instead of an error here.



```suggestion
            warnings.warn(
                "Support for labels represented as bytes is deprecated in v1.3 and will"
                " error in v1.5. Convert the labels to a string or integer format.",
                FutureWarning,
            )
```

The tests will need to be updated to check the warning instead of the `ValueError`.